src/typescript/ast-walker.ts

```diff
@@ -33,7 +34,18 @@ export const getImportsAndExports = (sourceFile: BoundSourceFile, options: Optio
       const filePath = module.resolvedModule.resolvedFileName;
       if (filePath) {
         if (module.resolvedModule.isExternalLibraryImport) {
-          if (module.resolvedModule.extension === '.d.ts') {
+          if (isSelfReferenceImport(workspace, specifier, filePath, module.resolvedModule.extension)) {
```

Comment:
Perhaps too naive, but maybe we could simplify things by checking for `isInNodeModules(filePath)` here?

Reply:
I've simplified it even more that that. We don't need to check where the actual file is, just how it is being imported.